test/suite/test_compact15.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wiredtiger
+import wttest
+from wtscenario import make_scenarios
+
+# test_compact15.py
+# This test checks that foreground compaction requires a URI.
+class test_compact15(wttest.WiredTigerTestCase):
+    uri = 'table:test_compact15'
+
+    uris = [
+        ('valid_uri', dict(valid_uri=True)),
+        ('invalid_uri', dict(valid_uri=False))
+    ]
+    scenarios = make_scenarios(uris)
+
+    def test_compact15(self):
+        # FIXME-WT-11399
+        if self.runningHook('tiered'):
+            self.skipTest("this test does not yet work with tiered storage")
+
+        self.session.create(self.uri)
+
+        if self.valid_uri:
+            self.session.compact(self.uri, None)
+        else:
+            self.assertRaisesWithMessage(wiredtiger.WiredTigerError, lambda:
+                self.session.compact(None, None),
```

Comment:
This can be added to the `test_compact06` which is there to "Test background compaction API usage". So we don't need another individual test.

Reply:
`test_compact06` is dedicated to bg compaction but we could use it for the foreground compact API as well?

Reply:
Ah I see, okay happy to leave as is then